react/PlaceDetails.tsx

```diff
@@ -0,0 +1,134 @@
+}
+
+const countryDescriptions: CountryDescription[] = [
+  {
+    name: 'ARG',
+    summary: [
+      [{ name: 'street' }, { delimiter: ' ', name: 'number' }],
+      [{ name: 'complement' }],
+      [{ name: 'postalCode' }],
+      [{ name: 'city' }, { delimiter: ', ', name: 'state' }],
+    ],
+  },
+  {
+    name: 'BRA',
+    summary: [
+      [
+        { name: 'street' },
+        { delimiter: ' ', name: 'number' },
+        { delimiter: ', ', name: 'complement' },
+      ],
+      [
+        { name: 'neighborhood', delimiterAfter: ' - ' },
+        { name: 'city' },
+        { delimiter: ' - ', name: 'state' },
+      ],
+      [{ name: 'postalCode' }],
+    ],
+  },
+  {
+    name: 'KOR',
+    summary: [
+      [{ name: 'street' }, { delimiter: ', ', name: 'complement' }],
+      [
+        { name: 'city' },
+        { delimiter: ', ', name: 'state' },
+        { delimiter: ' ', name: 'postalCode' },
+      ],
+    ],
+  },
+]
+
+const PlaceDetails: StorefrontFunctionComponent<PlaceDetailsProps> = () => {
+  const [option, setOption] = useState<string>('ARG')
+  const [summary, setSummary] = useState<LineComponent[][]>([])
+
+  return (
+    <div>
+      <Dropdown
+        label="Country"
+        options={[
+          { value: 'ARG', label: 'Argentina' },
+          { value: 'BRA', label: 'Brazil' },
+          { value: 'KOR', label: 'Korea' },
+        ]}
+        value={option}
+        onChange={(_: any, newVal: string) => {
+          setOption(newVal)
+          let newSummary = countryDescriptions.find(
+            description => description.name == newVal
+          )
+          setSummary(newSummary ? newSummary.summary : [])
+        }}
+        placeholder="Select a country"
+      />
+      {summary.map((line: LineComponent[], index: number) => [
+        ...line.map(
+          (field: LineComponent, index: number, line: LineComponent[]) => {
+            const hasPreviousField = index > 0 && address[line[index - 1].name]
+            const hasNextField =
+              index + 1 < line.length && address[line[index + 1].name]
+            const hasDifferentDelimiter = field.delimiterAfter !== '-'
+            const shouldShowDelimiter = hasNextField || hasDifferentDelimiter
+
+            return address[field.name] ? (
+              <span key={field.name}>
+                {field.delimiter && hasPreviousField && (
+                  <span className={field.name + '-delimiter'}>
+                    {field.delimiter}
+                  </span>
+                )}
+                <span className={field.name}>{address[field.name]}</span>
+                {field.delimiterAfter && shouldShowDelimiter && (
+                  <span className={field.name + '-delimiter-after'}>
+                    {field.delimiterAfter}
+                  </span>
+                )}
+              </span>
+            ) : null
+          }
+        ),
+        <br className={'line' + (index + 1) + '-delimiter'} key={index} />,
+      ])}
```

Comment:
This is really hard to read and understand what's going on. Break it into smaller variables/functions.

Reply:
I copied this part from the previous version and I didn't want to my significant changes in the logic until I know it won't break anything, having said that, I broke it down a bit, although it's still quite cumbersome. I plan to get back to this once I understand exactly how it works so that I can make a simpler version of this. Thanks!